src/libraries/System.Private.CoreLib/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -0,0 +1,164 @@
+        /// <param name="source">The span whose elements should be added to the <see cref="List{T}"/>.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="list"/> is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="index"/> is less than 0 or greater than <paramref name="list"/>'s <see cref="List{T}.Count"/>.</exception>
+        public static void InsertRange<T>(this List<T> list, int index, ReadOnlySpan<T> source)
+        {
+            if (list is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.list);
+            }
+
+            if ((uint)index > (uint)list._size)
+            {
+                ThrowHelper.ThrowArgumentOutOfRange_IndexMustBeLessOrEqualException();
+            }
+
+            if (!source.IsEmpty)
+            {
+                if (list._items.Length - list._size < source.Length)
+                {
+                    list.Grow(list._size + source.Length);
+                }
+
+                // If the index at which to insert is less than the number of items in the list,
+                // shift all items past that location in the list down to the end, making room
+                // to copy in the new data.
+                if (index < list._size)
+                {
+                    Array.Copy(list._items, index, list._items, index + source.Length, list._size - index);
+                }
+
+                // Copy the source span into the list.
+                // Note that this does not handle the unsafe case of trying to insert a CollectionsMarshal.AsSpan(list)
+                // or some slice thereof back into the list itself; such an operation has undefined behavior.
+                source.CopyTo(list._items.AsSpan(index));
+                list._size += source.Length;
+                list._version++;
+            }
+        }
+
+        /// <summary>Copies the entire <see cref="List{T}"/> to a span.</summary>
+        /// <typeparam name="T">The type of elements in the list.</typeparam>
+        /// <param name="list">The list from which the elements are copied.</param>
+        /// <param name="destination">The span that is the destination of the elements copied from <paramref name="list"/>.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="list"/> is null.</exception>
+        /// <exception cref="ArgumentException">The number of elements in the source <see cref="List{T}"/> is greater than the number of elements that the destination span can contain.</exception>
+        public static void CopyTo<T>(this List<T> list, Span<T> destination)
+        {
+            if (list is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.list);
+            }
+
+            new ReadOnlySpan<T>(list._items, 0, list._size).CopyTo(destination);
```

Comment:
Why not just `list._items.AsSpan()`?

Reply:
Because _size might be < list._items.Length.

Reply:
🤦 I missed that the second parameter wasn't `list._items.Length`